utils/logger.go

```diff
@@ -0,0 +1,109 @@
+// this is a new logger interface for mattermost
+
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"runtime"
+
+	l4g "github.com/alecthomas/log4go"
+)
+
+// this pattern allows us to "mock" the underlying l4g code when unit testing
+var debug = l4g.Debug
+var info = l4g.Info
+var err = l4g.Error
+
+// Logger - an instance of the log framework that can be used to log messages
+type Logger string
```

Comment:
The type is there for two reasons:
1. It gives me some place to put the name of the file that created the logger, for implicit inclusion in log messages.
2. It is the receiver type for all exported functions, which improves readability, because it lets you say `log.Info(ctx, "my message here")` instead of just `Info(ctx, "my message here")`, which is what you would get with static functions